autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -835,19 +843,17 @@ def file_output(
 
         # Abort if we don't want to output anything.
         if hasattr(self, 'disable_file_output'):
-            if self.disable_file_output:
+            if 'all' in self.disable_file_output:
                 return None, {}
-            else:
-                self.disabled_file_outputs = []
 
         # This file can be written independently of the others down bellow
-        if 'y_optimization' not in self.disabled_file_outputs:
+        if 'y_optimization' not in self.disable_file_output:
             if self.output_y_hat_optimization:
                 self.backend.save_targets_ensemble(self.Y_optimization)
 
         if hasattr(self, 'pipelines') and self.pipelines is not None:
```

Comment:
```suggestion
        if (
            'pipelines' in self.disable_file_output
            not hasattr(self, 'pipelines')
            or self.pipelines is None
            or self.pipelines[0] is None
            or len(self.pipelines) == 0
        ):
            pipelines = None
        elif self.task_type in CLASSIFICATION_TASKS:
            pipelines = VotingClassifier(estimators=None, voting='soft', )
            pipelines.estimators_ = self.pipelines
        elif self.task_type in REGRESSION_TASKS:
            pipelines = VotingRegressorWrapper(estimators=None)
            pipelines.estimators_ = self.pipelines
        else:
            raise ValueError(
                f'task_type must be in {REGRESSION_TASKS} or {CLASSIFICATION_TASKS}, '
                f'but got {self.task_type}'
            )
```

Reply:
I have refactored this also in the upcoming PR.

Reply:
Hey, could you separate L843--L867 in the old code as a new method in the upcoming PR?

Reply:
sure.